### ISSUE TITLE
Re-frame version?

### ISSUE THREAD

**Issue by pesterhazy** [pesterhazy/cljs-spa-example#17]
People have requested a version of this demo based on re-frame. @pedrorgirardi has made the necessary modificiations [in a branch](https://github.com/pesterhazy/cljs-spa-example/compare/master...pedrorgirardi:re-frame).

I'm undecided if changing the repo to use re-frame is a good idea:

- re-frame adds more boilerplate (whereas I'd like to keep the project small)
- it's not clear that re-frame, as opposed to vanilla Reagent, is necessary (or desirable) for small projects
- adding re-frame adds cognitive overhead

On the other hand re-frame is well documented, as well as a popular starting point for CLJS SPAs. Any thoughts?

**Comment by Juleffel** [comment]
I think, as you said, that re-frame is unnecessary for a small project and maybe shouldn't be included in the master version as a default choice. As you said, it's also not convenient for beginners. However, for people that use this as a starting point for bigger projects with reagent/double bundle, it would be good to have a solution, like for example a branch, or even a mentioned fork on the readme ?
Re-frame is really good for separation of concerns, pure functions, testing (unit tests and 10x) and so much more, and apparently the work is already done to integrate it to this repo, I think that would be a pity to not include it somehow.

**Comment by pedrorgirardi** [comment]
Hi there!

Yes @Juleffel, there's a **re-frame** version but it's a little bit outdated though. I like very much the design @pesterhazy has put together for this project and I'm not sure doing a **re-frame** version the way I did is a good thing. I made the changes to work with **re-frame** but I didn't think about its design. 

A **re-frame** specific 'template' with similiar ideas to this project is [kee-frame](https://github.com/ingesolvoll/kee-frame). Have you seen that?

I don't have much time to work on it now, but I could update the re-frame branch with the latest changes if you think it would be helpful.
